### PR TITLE
cli: fix generated clients (part 4)

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

File: pkg/oc/cli/describe/chaindescriber_test.go [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
 	"github.com/gonum/graph/concrete"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/sets"
-	kapi "k8s.io/kubernetes/pkg/api"
-
-	"github.com/openshift/origin/pkg/client/testclient"
-	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
 )
 
+/*
```
Comment (mfojtik) [inline pkg/oc/cli/describe/chaindescriber_test.go]: @fabianofranz are you willing to fix these? @deads2k suggest to create a list filtered by the scheme in the fake client. The describer and project status now takes 5 different clients for the resources it produces the status for.
The test is reading a fixture file with mixed resources and we need 5 different fake clients fed by appropriate objects to make this test pass.

File: pkg/oc/cli/cmd/request_project.go [hunk 2]
```diff
@@ -129,11 +132,22 @@ func (o *NewProjectOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, a
...
+				Resource: "projectrequests",
+			},
+		},
+	}
+	accessReviewResponse, err := o.SarClient.Create(accessReview)
 	if err != nil {
 		return err
 	}
+	if !accessReviewResponse.Status.Allowed {
+		return errors.New("you may not request a new project via this API")
```
Comment (deads2k) [inline pkg/oc/cli/cmd/request_project.go]: @mfojtik this is less informative than we used to receive.  What if you just hit the list endpoint manually?

Also, I'm really hoping this breaks an integration test.

File: pkg/oc/cli/cmd/startbuild_test.go [hunk 3]
```diff
@@ -250,8 +247,8 @@ func TestHttpBinary(t *testing.T) {
 		stdin := bytes.NewReader([]byte{})
 		stdout := &bytes.Buffer{}
 		options := buildapi.BinaryBuildRequestOptions{}
-		fakeclient := testclient.NewSimpleFake()
-		buildconfigs := FakeBuildConfigs{fakeclient.BuildConfigs("default"), t, tc.fromFile}
+		//fakeclient := buildfake.NewSimpleClientset()
```
Comment (deads2k) [inline pkg/oc/cli/cmd/startbuild_test.go]: remove cruft

File: pkg/oc/cli/describe/chaindescriber_test.go [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
+* FIXME: Disable because we need a better way to feed the multiple fake
```
Comment (deads2k) [inline pkg/oc/cli/describe/chaindescriber_test.go]: This should compile.  Can you just rename it or skip it?

Comment (deads2k) [reply pkg/oc/cli/describe/chaindescriber_test.go]: Oh, it won't compile, will it.....

File: pkg/oc/cli/cmd/request_project.go (hunk 2 above)
Comment (mfojtik) [reply pkg/oc/cli/cmd/request_project.go]: @deads2k i'm pretty sure the existing List() will return this error

**Comment by deads2k** [comment]
You'll need to spawn the p1 issues and they have to be worked.  The tests are particularly concerning.  I think I'll investigate to see how hard they are to fix with the scheme detection I described.

File: pkg/oc/cli/describe/chaindescriber_test.go (hunk 4 above)
Comment (mfojtik) [reply pkg/oc/cli/describe/chaindescriber_test.go]: @deads2k i will have to plumb the invocation of `NewChainDescriber` to take all clients that we need to make this compile so we can skip it, just don't see the point if we are going to rework the entire test anywayt.

Comment (deads2k) [reply pkg/oc/cli/describe/chaindescriber_test.go]: > @deads2k i will have to plumb the invocation of NewChainDescriber to take all clients that we need to make this compile so we can skip it, just don't see the point if we are going to rework the entire test anywayt.

I doubt we refactor the entire test.  More likely a filter.

**Comment by mfojtik** [comment]
@deads2k tests fixed and manual get to project requests added

File: pkg/image/apis/image/helper.go [hunk 5]
```diff
@@ -2,21 +2,23 @@ package image
...
-	"k8s.io/apimachinery/pkg/api/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 
 	"github.com/blang/semver"
 	"github.com/golang/glog"
 
 	"github.com/openshift/origin/pkg/image/apis/image/internal/digest"
+	quotautil "github.com/openshift/origin/pkg/quota/util"
```
Comment (deads2k) [inline pkg/image/apis/image/helper.go]: This shouldn't be allowed.  Did the verify-imports slip?

File: pkg/oc/cli/cmd/request_project.go [hunk 6]
```diff
@@ -31,7 +32,8 @@ type NewProjectOptions struct {
 
 	SkipConfigWrite bool
 
-	Client client.Interface
+	Client    projectclient.ProjectInterface
+	SarClient authorizationclient.SelfSubjectAccessReviewInterface
```
Comment (deads2k) [inline pkg/oc/cli/cmd/request_project.go]: dead?

File: pkg/oc/cli/cmd/startbuild.go [hunk 7]
```diff
@@ -560,12 +563,12 @@ func streamPathToBuild(repo git.Repository, in io.Reader, out io.Writer, client
 			// NOTE: It's important that this stays false unless we change the
 			// path to something else, otherwise we will delete whatever path the
 			// user provided.
-			var usedTempDir bool = false
-			var tempDirectory string = ""
+			var usedTempDir bool
```
Comment (deads2k) [inline pkg/oc/cli/cmd/startbuild.go]: nit, I prefer the old code.

**Comment by deads2k** [comment]
I'm going to stick on the import problem.  images shouldn't depend on quota.

**Comment by deads2k** [comment]
> I'm going to stick on the import problem. images shouldn't depend on quota.

otherwise, lgtm

File: pkg/oc/cli/cmd/startbuild.go (hunk 7 above)
Comment (mfojtik) [reply pkg/oc/cli/cmd/startbuild.go]: @deads2k golint will haunt you

Comment (deads2k) [reply pkg/oc/cli/cmd/startbuild.go]: > @deads2k golint will haunt you

golint is wrong.

File: pkg/image/apis/image/helper.go (hunk 5 above)
Comment (mfojtik) [reply pkg/image/apis/image/helper.go]: no it catched it, I will copy it into image import as that is the only place that uses it and make a private copy in legacy client as that is going away anyway

File: pkg/build/client/internalversion/buildinstantiatebinary.go [hunk 8]
```diff
@@ -28,8 +28,8 @@ func (c *buildInstatiateBinary) InstantiateBinary(name string, options *buildapi
 		Namespace(c.ns).
 		Resource("buildconfigs").
 		Name(name).
-		SubResource("instantiateBinary").
-		Body(c).
+		SubResource("instantiatebinary").
```
Comment (mfojtik) [inline pkg/build/client/internalversion/buildinstantiatebinary.go]: @deads2k i never know that this will make difference in the RBAC rules...

**Comment by deads2k** [comment]
@mfojtik failures are real, but you only have two!  Update the field selectors.

**Comment by mfojtik** [comment]
\o/

**Pull request merged**